x11rb-protocol/src/errors.rs

```diff
@@ -65,3 +69,89 @@ impl std::fmt::Display for ParseError {
         }
     }
 }
+
+/// An error that occurred while connecting to an X11 server
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum ConnectError {
+    /// An unknown error occurred.
+    ///
+    /// One situation were this error is used when libxcb indicates an error that does not match
+    /// any of the defined error conditions. Thus, libxcb is violating its own API (or new error
+    /// cases were defined, but are not yet handled by x11rb).
+    UnknownError,
+
+    /// Error while parsing some data, see `ParseError`.
+    ParseError(ParseError),
+
+    /// Out of memory.
+    ///
+    /// This is `XCB_CONN_CLOSED_MEM_INSUFFICIENT`.
+    InsufficientMemory,
+
+    /// Error during parsing of display string.
+    ///
+    /// This is `XCB_CONN_CLOSED_PARSE_ERR`.
+    DisplayParsingError,
+
+    /// Server does not have a screen matching the display.
+    ///
+    /// This is `XCB_CONN_CLOSED_INVALID_SCREEN`.
+    InvalidScreen,
+
+    /// An I/O error occurred on the connection.
+    IoError(std::io::Error),
```

Comment:
This is I/O-y. Is it okay to have that in this crate? :-/

I'd prefer to leave this in x11rb, but... hm...